crates/ruff_index/src/list.rs

```diff
@@ -232,3 +233,18 @@ impl<I: Idx, K: Clone + Ord, V: Clone> ListBuilder<I, K, V> {
         self.add_cell(new_key, new_value, new_tail)
     }
 }
+
+impl<I: Idx, K: Clone, V> ListBuilder<I, K, V> {
```

Comment:
Nit: I'd probably move the `Clone` constraint closer to `map` to avoid that we accidentally "inherit" it for other methods adding to this `ListBuilder` impl block